### ISSUE TITLE
Unable to reproduce evaluation results from the paper

### ISSUE THREAD

**Issue by Nekroz05** [lpiccinelli-eth/UniDepth#71]
I'm trying to reproduce the evaluation results presented in the paper but I'm encountering difficulty. I'd appreciate guidance on how to accurately replicate the results.

```
        model = UniDepthV1.from_pretrained("lpiccinelli/unidepth-v1-vitl14")

        # Move to CUDA, if any
        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
        self.model = model.to(device)
```

I'm evaluating on the KITTI eigen Split and NYUv2 Official Split
I'm using both v1 and v2 with vitl14
i'm using intrinsic matrix for inference

![Screenshot from 2024-09-09 14-48-49](https://github.com/user-attachments/assets/f7c6c84c-91c1-4d95-b010-9be7ab46c43b)

Question
1. What specific configuration was used to achieve the results reported in the paper?
2. Are there any additional parameters or settings I should be aware of?
3. Is there a specific evaluation script I should be using?
4. Are there any post-processing steps applied to the depth predictions?

My Current Result
although not way off, it isn't exactly the same either, especially on RMSE

NYUv2
![image](https://github.com/user-attachments/assets/4386ced6-0dba-40f2-9a7c-66a25d20bbae)

KITTI
![image](https://github.com/user-attachments/assets/9bc65c01-d999-4ab4-a587-cddf15955662)

**Comment by lpiccinelli-eth** [comment]
Dear @Nekroz05 , sorry for the late reply. You can check the `validation.py` file under `unidepth/utils` in the `training` branch (PR #76) and also `base_dataset.py` `accumulate_metrics` method to see how it is carried out. 

However, I suspect there may be some rescaling of both prediction or GT since the most affected metric is RMSE, while the metrics where the scale cancels out are pretty consistent.

**Comment by Nekroz05** [comment]
I'll be sure to check it out, Thank you for your response!

**Comment by Cats-eat-peppercorns** [comment]
Hi @Nekroz05 

Sorry for commenting on a closed issue from months ago. I'm trying to reproduce the NYUv2 fine-tuning results for UniDepthV2 (Table V in the paper, δ1=98.8%), but I'm stuck and could use some guidance.

My situation:

I'm using the pretrained lpiccinelli/unidepth-v2-vitl14 checkpoint from HuggingFace.
Zero-shot on NYUv2: δ1 ≈ 95% — which seems unusually low for a V2 model.
After fine-tuning on NYUv2 (795 train / 654 test): δ1 ≈ 97.3% — no matter what I try, I can't break past this ceiling.
I've tried various learning rates (1e-5 to 1e-4), cosine annealing vs constant lr, disabling camera loss (weight=0), different warmup settings, and EMA configurations. The result stays at ~97.3%.
I noticed you reported UniDepthV1 ViT-L zero-shot δ1=0.982 on NYUv2 (654 test images). That's significantly higher than my V2 zero-shot (~95%), which doesn't make sense since V2 should be better than V1.

Questions:

Which checkpoint did you use? Did you load from lpiccinelli/unidepth-v1-vitl14 on HuggingFace, or a different one?
Evaluation protocol: Did you use the standard Eigen crop [45:471, 41:601] with depth range [[0.005, 10.0]m]? Did you use GT camera intrinsics at inference?
Image input size: Did you resize/pad to a specific resolution, or use the original 480×640?
I've verified my eval mask (45:-9, 41:-39), depth range (0.005-10.0m), and [depth_scale=1000]. Everything looks correct. The 3% gap between my zero-shot (95%) and yours (98.2%) on the same dataset is very puzzling.

Any help would be greatly appreciated!
